server/githookHandler.go

```diff
@@ -0,0 +1,84 @@
+package server
+
+import (
+	"bytes"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/eventgrid/2018-01-01/eventgrid"
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/go-autorest/autorest/date"
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/gobuffalo/buffalo"
+	"github.com/gobuffalo/buffalo/render"
+	"github.com/google/go-github/github"
+	"github.com/google/uuid"
+)
+
+// GithookReceiver converts a Github event into an Eventgrid event and publishes the event to Eventgrid
+func GithookReceiver(c buffalo.Context) error {
+	webhookSecret := os.Getenv("webhookSecret")
+	eventgridSecret := os.Getenv("eventgridSecret")
+	eventgridEndpoint := os.Getenv("eventgridEndpoint")
+
+	request := c.Request()
+
+	// save request body so it can be read multiple times
+	var bodyBytes []byte
+
+	if request.Body != nil {
+		bodyBytes, _ = ioutil.ReadAll(request.Body)
+	}
+
+	request.Body = ioutil.NopCloser(bytes.NewBuffer(bodyBytes))
+
+	// validate githook using secret
+	_, err := github.ValidatePayload(request, []byte(webhookSecret))
+
+	if err != nil {
+		log.Printf("Validation failed: err=%s\n", err)
+		return c.Error(http.StatusInternalServerError, err)
+	}
+
+	// parse webhook body
+	defer request.Body.Close()
```

Comment:
Because this comes after the body has been reassigned, this will just No-Op after your function returns. I'd skip the defer and just call request.Body.Close() after you call [`io.LimitReader`](https://godoc.org/io#LimitReader)